DietPi-Globals | G_RPI_UPDATE: Added APT hold for "libraspberrypi0", to hold complete "raspberrypi-firmware" package list

MichaIng opened the pull request:
Ref: https://github.com/Hexxeh/rpi-update, https://github.com/raspberrypi/firmware
- Should we let rpi-update remove old modules directories automatically or keep them in case of failure or unstable firmware?

Note:
- If one chooses to uninstall the firmware packages before (makes sense, removes docs and other unnecessary stuff, all 4 packages are well overwritten/replaced by rpi-update!), then `apt-mark hold ...` behaves a bid strange for not installed packages, leading to wrong/not intended hold states: https://bugs.launchpad.net/ubuntu/+source/apt/+bug/1694989
- But this works fine for installed packages, thus should be no issue for us and corrected as fast as APT packages are installed and hold state becomes relevant again.

Commit:
v6.6

+ DietPi-Globals | G_RPI_UPDATE: Added APT hold for "libraspberrypi0", to hold complete "raspberrypi-firmware" package list

Fourdee commented:
@MichaIng 

> Should we let rpi-update remove old modules directories automatically or keep them in case of failure or unstable firmware?

We tried this in the past with patching our custom kernel (unzipped), and, ```dietpi-config```, which removed ```/lib/modules/*``` prior to run of ```rpi-update```. But some users had a failed update, rebooted and then missing modules.

Since then, we disabled the auto removal of current modules. As much as I love auto cleaning of old modules, its not worth the risk for end users in case of a failure.
PREP being the exclusion to this.

Fourdee commented on dietpi/func/dietpi-globals:
```diff
@@ -1416,10 +1416,8 @@ $print_logfile_info
-			export SKIP_WARNING=1
-			G_RUN_CMD rpi-update
-			apt-mark hold raspberrypi-bootloader raspberrypi-kernel libraspberrypi-bin
+			SKIP_BACKUP=1 SKIP_WARNING=1 G_RUN_CMD rpi-update #PRUNE_MODULES=1 would remove old/unused /lib/modules/* directories, too dangerous if update fails/firmware is unstable?
```
Nice 👍 

I did try this, but after ```G_RUN_CMD SKIP_BACKUP=1 SKIP_WARNING=1```, thats why it failed 😄

MichaIng replied:
@Fourdee 
Jep, also tried it the other way round first, then also was not sure if variables are forwarded through G_RUN_CMD to sub command, but it works 😃.

Fourdee reviewed (approved):
👍 Legend

Commit:
v6.6

+ DietPi-Globals | G_RPI_UPDATE: Added APT hold for "libraspberrypi0", to hold complete "raspberrypi-firmware" package list

Commit:
Merge branch 'testing' into rpi_update_hold_libraspberrypi0
